src/page-objects/administration/RuleDetail.ts

```diff
@@ -39,6 +43,15 @@ export class RuleDetail extends RuleCreate implements PageObject {
         this.promotionCustomerRulesCardEmptyState = this.promotionCustomerRulesCard.getByRole('alert');
         this.promotionCartRulesCard = page.locator('.sw-settings-rule-detail-assignments__card-promotion_cart_rule');
         this.promotionCartRulesCardEmptyState = this.promotionCartRulesCard.getByRole('alert');
+        if (satisfies(instanceMeta.version, '<6.7')) {
+            this.assignmentModalSearchField = page.locator('.sw-settings-rule-add-assignment-modal').getByPlaceholder('Search...');
+            this.assignmentModalAddButton = page.locator('.sw-button--primary').getByText('Add');
+            this.shippingMethodAvailabilityRulesCardSearchField = this.shippingMethodAvailabilityRulesCard.getByPlaceholder('Search...');
+        } else {
+            this.assignmentModalSearchField = page.locator('.sw-settings-rule-add-assignment-modal').getByRole('textbox');
+            this.assignmentModalAddButton = page.locator('.mt-button--primary').getByText('Add');
+            this.shippingMethodAvailabilityRulesCardSearchField = this.shippingMethodAvailabilityRulesCard.getByRole('textbox');
```

Comment:
getByRole('textbox') should work for both 6.6 and 6.7 right?
only change we need is **.mt-button--primary**?

Reply:
This should also work. Please don't use `getByPlaceholder()` if possible.
```suggestion
            this.assignmentModalSearchField = page.locator('.sw-settings-rule-add-assignment-modal').getByRole('textbox');
            this.shippingMethodAvailabilityRulesCardSearchField = this.shippingMethodAvailabilityRulesCard.getByRole('textbox');
             if (satisfies(instanceMeta.version, '<6.7')) {
                 this.assignmentModalAddButton = page.locator('.sw-button--primary');
             } else {
                 this.assignmentModalAddButton = page.locator('.mt-button--primary');  
```

Reply:
Thank you, you both are totally right. I have pushed a new version. ✔️